sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2035,22 +2055,7 @@ device_image_plain ProgramManager::build(const device_image_plain &DeviceImage,
     if (!DeviceCodeWasInCache &&
         InputImpl->get_bin_image_ref()->supportsSpecConstants()) {
       enableITTAnnotationsIfNeeded(NativePrg, Plugin);
-
-      std::lock_guard<std::mutex> Lock{InputImpl->get_spec_const_data_lock()};
-      const std::map<std::string,
-                     std::vector<device_image_impl::SpecConstDescT>>
-          &SpecConstData = InputImpl->get_spec_const_data_ref();
-
-      for (const auto &DescPair : SpecConstData) {
-        for (const device_image_impl::SpecConstDescT &SpecIDDesc :
-             DescPair.second) {
-          if (SpecIDDesc.IsSet) {
-            Plugin.call<PiApiKind::piextProgramSetSpecializationConstant>(
-                NativePrg, SpecIDDesc.ID, SpecIDDesc.Size,
-                SpecConsts.data() + SpecIDDesc.BlobOffset);
-          }
-        }
-      }
+      setSpecializationConstants(InputImpl, NativePrg, Plugin);
```

Comment:
Should we fold `enableITTAnnotationsIfNeeded(NativePrg, Plugin)` into the helper as well?

Reply:
We cannot fully merge it as it is used in another place that sets specialization constants differently. However, we may as well call it inside `setSpecializationConstants`.